src/puya/ir/builder/main.py

```diff
@@ -373,9 +373,24 @@ def visit_decimal_constant(self, expr: awst_nodes.DecimalConstant) -> TExpressio
                 )
 
     def visit_bool_constant(self, expr: awst_nodes.BoolConstant) -> TExpression:
-        return UInt64Constant(
-            value=int(expr.value), ir_type=IRType.bool, source_location=expr.source_location
-        )
+        match expr.wtype:
+            case wtypes.bool_wtype:
+                return UInt64Constant(
+                    value=int(expr.value),
+                    ir_type=IRType.bool,
+                    source_location=expr.source_location,
+                )
+            case wtypes.arc4_bool_wtype:
+                return BytesConstant(
+                    value=(128 if expr.value else 0).to_bytes(1),
```

Comment:
It would be nice to keep ARC4 specific logic within the arc4 builder module, but not sure if it's worth it for this small bit of code. Maybe just have an ARC4_FALSE and ARC4_TRUE `bytes` constant in that module and use them here?